authlib/integrations/httpx_client/utils.py

```diff
@@ -1,5 +1,8 @@
 from httpx import URL
-from httpx.content_streams import ByteStream
+try:
+    from httpx._content_streams import ByteStream
```

Comment:
I don't think it is a good idea to import from private modules. I need to ask @encode team to make it public.

Reply:
So, it's not obvious to me that the `rebuild_request` function needs to be written in a way that uses private API here. Can you change that function instead? Perhaps like so...

```
def rebuild_request(request, url, headers, body):
    method = request.method
    headers = Headers(request.headers)
    headers.update(headers)
    return Request(method=method, url=url, headers=headers, body=body)
```

Reply:
@tomchristie unfortunately, there is no `body` parameter in `Request`.

Reply:
Let me rephrase that. `content=...`

Reply:
@tomchristie 

```
class Request:
    def __init__(
        self,
        method: str,
        url: typing.Union[str, URL],
        *,
        params: QueryParamTypes = None,
        headers: HeaderTypes = None,
        cookies: CookieTypes = None,
        data: RequestData = None,
        files: RequestFiles = None,
        json: typing.Any = None,
        stream: ContentStream = None,
    ):
```

This is the parameters of `Request`, there is no suitable parameter for `body`.

Reply:
@tomchristie there is no `content` either, the only available parameter would be `data=` (I think). But this `rebuild_request` is used in `auth_flow`, and it is `requires_request_body = True`. The `request` returned by `rebuild_request` is not `read` or `aread` yet, thus this `request` would not have `.content` property.

Reply:
@tomchristie it would also be helpful if we can add a property setter on `Request`:

```py
    @content.setter
    def content(self, value: bytes) -> None:
        self._content = value
        self.stream = ByteStream(self._content)
```

Do you think it is ok to add this setter?